src/main.lua

```diff
@@ -1,12 +1,21 @@
 require("class")
 local StackManager = require("stackManager")
+local enabledFunctions = require("gamestates/enabledFunctions")
 local MainMenu = require("gamestates/mainMenu")
 
 local state
 
 function love.load()
 	state = StackManager.create(MainMenu, "stackQueue")
 
+	local function getStateFunction(t, key)
+		if enabledFunctions[key] then
+			return state[key]
+		end
+	end
+
+	setmetatable(love, {__index = getStateFunction})
```

Comment:
Oh boy. I have a couple concerns:

1. Does this actually work? I don't have complete mastery of closures in Lua, but I'm worried that `getStateFunction()` might have a copy of `state` that doesn't update when the state variable updates.
2. Can you make it *super* obvious that this causes `love.mousepressed`, `love.mousereleased`, etc. to be called? One possible idea: you could rename `key` to `eventFunction`, then it would read `local function getStateFunction(t, eventFunction)`. Or maybe `loveEventFunction` or `loveEvent`. Hm...
3. What kind of error do you get if you call a function in love that doesn't exist? For example, what's the error that displays when you do `love.blah()`?

Reply:
1: yes it works
2: This was something I threw together, because it was simpler than what we discussed.
 I am open to renaming things.
3: I haven't tried yet. I didn't feel the need to sugar coat it. I just wanted it working and get your feed back.

Reply:
3: extended it does throw any error if you set InGame.draw as InGame.drawa I just doesn't work.
love.blah()???? Where would you be calling this? is it love2d defined or user-defined?

Reply:
Ok, so it doesn't really make it harder to debug. It doesn't hide the fact that you called a function that doesn't exist, and you don't lose track of where the mistake happened.